app/assets/javascripts/live_search.js

```diff
@@ -45,73 +50,94 @@
     this.focusErrorMessagesOnLoad(this.$form)
 
     // prevent page refresh on search submit button click
-    // instead trigger the AJAX results fetch
-    this.$searchSubmitButton.on('click',
-      function (e) {
+    // instead trigger the results fetch
+    if (this.$searchSubmitButton) {
+      this.$searchSubmitButton.addEventListener('click', function (e) {
         e.preventDefault()
         this.formChange(e)
-      }.bind(this)
-    )
+      }.bind(this))
+    }
 
     if (GOVUK.support.history()) {
       this.saveState()
 
-      this.$form.on('change', 'input[type=checkbox], input[type=radio], select',
-        function (e) {
-          this.formChange(e)
-        }.bind(this)
-      )
+      var checkboxRadioSelects = this.$form.querySelectorAll('input[type=checkbox], input[type=radio], select')
```

Comment:
No, they all come from a Rails template so that the non-JS version still has all the same options. JS is used to show/hide/disable/select/unselect some of them on page load. Is that okay? I didn't fully understand what event you were referring to.

Reply:
I understand now. That's a very good point which I hadn't considered, but fortunately for me the form elements don't get updated when the search changes - only the results portion of the page is changed.

However, it turns out I was being needlessly specific - I can just set the 'change' eventListener on the form itself, and the event bubbles up. Code's a lot simpler, only one event listener set instead of hundreds. Thanks!